src/talos/moi_tests.py

```diff
@@ -101,10 +101,12 @@ def check_for_second_hit(
     if sample not in comp_hets:
         return []
 
-    partners = comp_hets[sample].get(first_variant, [])
-    if require_non_support:
-        return [partner for partner in partners if partner.sample_category_check(sample, allow_support=False)]
-    return partners
+    # thin out the possible partners by alt depth
+    return [
+        var
+        for var in comp_hets[sample].get(first_variant, [])
+        if not var.check_minimum_alt_depth(sample, min_alt_depth)
+    ]
```

Comment:
I think this was the issue - I was getting a Bool value for if the current variant was support-only, then using that to decide if a partner could also be support-only. It was a mess of booleans, so I've dropped it from here completely. Once we find all the partners, there's a simpler test on each pair that at least one must be non-support.